src/ast/ast_validations.c

```diff
@@ -474,11 +507,19 @@ static AST_Validation _Validate_MATCH_Clauses(const AST *ast, char **reason) {
 		AST_ReferredFunctions(match_clause, referred_funcs);
 	}
 
+	// Verify that no relation alias is also used to denote a node
+	for(uint i = 0; i < match_count; i ++) {
+		const cypher_astnode_t *match_clause = match_clauses[i];
+		res = _ValidateReusedIdentifiers(identifiers, cypher_ast_match_get_pattern(match_clause), reason);
```

Comment:
It's a bit difficult to understand that, at this point `identifiers` contains only edge aliases.